### PR TITLE
[data] Also improve repr of pandas dtype

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

This applies https://github.com/ray-project/ray/pull/34286 to the pandas extension dtype as well.

### CODE REVIEW

File: python/ray/air/util/tensor_extensions/pandas.py [hunk 1]
```diff
@@ -439,7 +439,7 @@ def __from_arrow__(self, array: Union[pa.Array, pa.ChunkedArray]):
         return TensorArray(values)
 
     def __str__(self) -> str:
-        return self.name
+        return self.name.replace("TensorDtype", "numpy.ndarray")
```
Comment (ericl) [inline python/ray/air/util/tensor_extensions/pandas.py:442]: Seems like the name is magically assigned.

**Review by amogkam — APPROVED** [review]

**Review by pcmoritz — APPROVED** [review]
Thanks for fixing this :)

**Comment by scv119** [comment]
looks this break the data:test_strict_mode.
image

**Comment by ericl** [comment]
Unfortunate PR collision--- Fixed in master.

**Pull request merged**